great_expectations/data_context/store/_store_backend.py

```diff
@@ -216,6 +221,10 @@ def _validate_value(self, value) -> None:  # noqa: B027 # no abstract decorator
     def _get(self, key) -> None:
         raise NotImplementedError
 
+    @abstractmethod
+    def _get_all(self) -> list[Any]:
+        raise NotImplementedError
```

Comment:
`#future-enhancement`

Can we make the StoreBackend Generic and then we don't have to use `Any` for the return types?

https://docs.python.org/3/library/typing.html#user-defined-generic-types

```python
from typing import TypeVar, Generic

T = TypeVar("T")

class StoreBackend(Generic[T], metaclass=ABCMeta):

    def get_all(self) -> T:
        return self._get_all()

    @abstractmethod
    def _get_all(self) -> list[T]:
        raise NotImplementedError

````

Reply:
Hmm I agree with this but we're not actually returning hydrated objects (just dict configs) - we should be returning rich config types or actual datasources but our stores use primitives